packages/shared/src/sentry.ts

```diff
@@ -0,0 +1,126 @@
+import { AuthInfo } from "./graphql/fetchAuthInfoFromGraphQL";
+import { getDeviceId } from "./getDeviceId";
+import { randomUUID } from "crypto";
+import { formatTags, logger, Tags } from "./logger";
+
+const SENTRY_DSN =
+  "https://5c145b72bb502832982243d6584f163d@o437061.ingest.us.sentry.io/4507534763819008"; // write-only permissions
+
+class SentryAPI {
+  private deviceId: string;
+  private sessionId: string;
+  private authInfo?: AuthInfo;
+  private initialized: boolean = false;
+
+  constructor() {
+    this.deviceId = getDeviceId();
+    this.sessionId = randomUUID();
+  }
+
+  initialize(app: string, version: string | undefined) {
+    if (this.initialized) {
+      console.warn(`Logger already initialized.`);
+    }
+
+    this.initSentry(app, version);
+    this.initialized = true;
+  }
+
+  private initSentry(app: string, version: string | undefined) {
+    Sentry.init({
+      dsn: SENTRY_DSN,
+      integrations: [nodeProfilingIntegration()],
+    });
+
+    Sentry.setTags({
+      app,
+      version,
+    });
+  }
+
+  identify(authInfo: AuthInfo) {
+    this.authInfo = authInfo;
+  }
+
+  async close() {
+    if (process.env.REPLAY_TELEMETRY_DISABLED) {
+      return;
+    }
+
+    await Sentry.close();
+  }
+
+  async flush() {
+    if (process.env.REPLAY_TELEMETRY_DISABLED) {
+      return;
+    }
+
+    await Sentry.flush();
+  }
+
+  captureException(error: Error, tags?: Tags) {
+    if (process.env.REPLAY_TELEMETRY_DISABLED) {
+      return;
+    }
+
+    const formattedTags = formatTags(tags);
+
+    const entry: Record<string, any> = {
+      ...formattedTags,
+      deviceId: this.deviceId,
+      sessionId: this.sessionId,
+    };
+
+    if (this.authInfo) {
+      switch (this.authInfo.type) {
+        case "user": {
+          entry.userId = this.authInfo.id;
+          break;
+        }
+        case "workspace": {
+          entry.workspaceId = this.authInfo.id;
+          break;
+        }
+      }
+    }
+
+    Sentry.captureException(error, entry);
+  }
+}
+
+export const sentry = new SentryAPI();
+
+// This should be called with the name once at the entry point.
+// For example, with the Playwright plugin, it is called in the Reporter interface constructor.
+function initSentry(app: string, version: string | undefined) {
+  sentry.initialize(app, version);
+}
+
+async function withSentry<T>(
+  methodName: string,
+  fn: () => T | Promise<T>,
+  tags?: Tags
+): Promise<T> {
+  try {
+    const result = await fn();
+    return result;
+  } catch (error: any) {
+    logger.error(`${methodName}Failed`, tags);
+    sentry.captureException(error);
+    throw error;
+  }
+}
+
+function withSentrySync<T>(methodName: string, fn: () => T): T {
+  try {
+    return fn();
+  } catch (error: any) {
+    logger.error(`${methodName}:Failed`, error);
+    sentry.captureException(error);
+    throw error;
+  }
+}
+
+export { initSentry, withSentry, withSentrySync };
```

Comment:
Small nit: I prefer exports that are inlined (e.g. `export function foo() {}`) because I think it's more readable. (When I'm looking at a function, I can tell if it's "private" or exported without having to grep elsewhere in the file.

Reply:
Thanks, that's a good recommendation. I agree that making the scope and visibility of functions, methods, and classes explicit at the point of declaration is more readable.